node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -213,6 +221,23 @@ impl<C: Context> ActiveValidator<C> {
         vec![Effect::NewVertex(ValidVertex(Vertex::Vote(proposal_vote)))]
     }
 
+    /// Registers the finalization of a block and checks whether the round exponent should be
+    /// changed
+    pub(crate) fn handle_finalized_block(&mut self, block_timestamp: Timestamp) {
+        match self
+            .round_success_meter
+            .handle_finalized_block(block_timestamp, Timestamp::now())
```

Comment:
If there's many finalized blocks, `Timestamp::now()` will keep changing. Shouldn't we be using the same timestamp for all finalized blocks?

Reply:
Oh, if we finalize multiple blocks at once (which can happen) – will we decrease the round exponent once or multiple times?

Reply:
Uhh, good point. I imagined we would change the exponent once and reset the "meter"... but the way it is coded now, we would probably change it multiple times, because the resetting of the meter happens when we create a new vote (which will probably not happen between handling of multiple finalized blocks). I need to fix that :grimacing:

Reply:
As for the timestamp, I don't think it matters that much, but I'll probably change this when fixing the multiple changes issue.